### PR TITLE
network/dhcp-server: introduce PersistLeases= setting

### PR DESCRIPTION
**Pull request by yuwata**
Requested at https://github.com/systemd/systemd/pull/31772#issuecomment-2000053357.

### CODE REVIEW

**Review by bluca — APPROVED** [review]

File: src/network/networkd-network-gperf.gperf [hunk 1]
```diff
@@ -350,6 +350,7 @@ DHCPServer.BootServerAddress,                config_parse_in_addr_non_null,
 DHCPServer.BootServerName,                   config_parse_dns_name,                                    0,                             offsetof(Network, dhcp_server_boot_server_name)
 DHCPServer.BootFilename,                     config_parse_string,                                      CONFIG_PARSE_STRING_SAFE_AND_ASCII, offsetof(Network, dhcp_server_boot_filename)
 DHCPServer.RapidCommit,                      config_parse_bool,                                        0,                             offsetof(Network, dhcp_server_rapid_commit)
+SHCPServer.PersistLeases,                    config_parse_tristate,                                    0,                             offsetof(Network, dhcp_server_persist_leases)
```
Comment (evverx) [inline src/network/networkd-network-gperf.gperf:353]: Looks like `networkd` just crashes
```
Failed to start systemd-networkd.service.
systemd-networkd.service: Scheduled restart job, restart counter is at 4.
Starting systemd-networkd.service...
/etc/systemd/networkd.conf:36: Unknown section 'DHCPServer'. Ignoring.
lo: Link UP
lo: Gained carrier
host0: Configuring with /usr/lib/systemd/network/80-container-host0.network.
free(): invalid pointer
systemd-networkd.service: Main process exited, code=dumped, status=6/ABRT
systemd-networkd.service: Failed with result 'core-dump'.
Failed to start systemd-networkd.service.
systemd-networkd.service: Scheduled restart job, restart counter is at 5.
systemd-networkd.service: Start request repeated too quickly.
systemd-networkd.service: Failed with result 'core-dump'.
Failed to start systemd-networkd.service.
```
I'm not sure why the CI didn't catch this.

Comment (evverx) [reply src/network/networkd-network-gperf.gperf:353]: The "invalid free" thing came from https://github.com/systemd/systemd/pull/30952 but `SHCPServer` doesn't look right anyway :-)

Comment (yuwata) [reply src/network/networkd-network-gperf.gperf:353]: Ouch. Will fix soon.

Comment (yuwata) [reply src/network/networkd-network-gperf.gperf:353]: -> #31994.

**Pull request merged**